Feature/v5 react themes

ektasawant opened the pull request:
<!-- If this pull request fixes an Issue, link it below. If not, you can remove the line below -->
Fixes 2728
Issue link [30](https://github.com/brightlayer-ui/react-themes/issues/30)
<!-- Include a bulleted list summarizing the main changes you have made in this PR -->
#### Changes proposed in this Pull Request:

- [x] Migrate to MUIv5
- [x] Update Shadow CSS for Contained buttonToggleGroup on hover.
- [x] Update bg color for valueLabelIcon valueLabelIcon
- [x] Remove css for wrapper from tab https://github.com/mui/material-ui/pull/26926
- [x] Update CSS for MuiAppbar dark theme.
- [x] Remove badge overrides css from dark theme.

<!-- Include screenshots if they will help illustrate the changes in this PR -->
#### Screenshots:
<img width="1778" alt="Screenshot 2022-03-02 at 5 40 34 PM" src="https://user-images.githubusercontent.com/10433274/156359233-0393019c-1b12-4209-ab69-97d8f2f408d0.png">

<img width="1642" alt="Screenshot 2022-03-02 at 5 40 21 PM" src="https://user-images.githubusercontent.com/10433274/156359257-517c9f71-bdd8-4f27-94ef-b7a925bc1f5c.png">


<!-- Instruction for PR reviewers, if more complicated than a simple yarn start -->
#### To Test:
- 
- cd demos/showcase and run git checkout mui-v5
- Run yarn start .
Note: Warning is known issue. https://github.com/styled-components/stylis-plugin-rtl/issues/33

joebochill commented on package.json:
```diff
@@ -34,13 +34,17 @@
         "@fontsource/open-sans": "^4.2.2",
-        "color": "^3.1.3"
+        "color": "^3.1.3",
+        "@mui/material": "^5.4.0",
+        "@mui/styles": "^5.3.0"
```
Are these actually dependencies, or do they just need to be peerDependencies (I'm thinking probably the latter)

joebochill commented on src/blueDarkTheme.ts:
```diff
@@ -53,14 +54,14 @@ const Spacing = 8;
     to customize in our themes. These have changed periodically from version to version of Material UI.
 */
-export const blueDarkTheme: ThemeOptions = {
+export const blueDarkTheme = createTheme({
```
this is fundamentally changing how our theme gets used (instead of just exporting the options and letting users call createTheme, we're now exporting the entire generated theme). Do we need to be doing this? How does this affect a user's ability to override/adjust theme values?

ektasawant replied:
Overriding works. Please review https://codesandbox.io/s/material-demo-forked-b1h8ps?file=/demo.js.

joebochill commented on src/blueDarkTheme.ts:
```diff
@@ -73,938 +74,992 @@ export const blueDarkTheme: ThemeOptions = {
+                    color: BLUIColors.lightBlue[300],
+                },
+                track: {
+                    height: 6,
```
looks like we removed a lot of styles around the slider in here...does this still look the way we expect?

ektasawant replied:
after migrating to v5 , observed deleted code is not necessary. CSS must have fix in v5.

MUI v4:
<img width="417" alt="Screenshot 2022-03-07 at 5 52 58 PM" src="https://user-images.githubusercontent.com/10433274/157034090-2b267f5d-15b2-40ab-aa72-83d2c9abe3a3.png">

MUI v5:
<img width="424" alt="Screenshot 2022-03-07 at 5 52 49 PM" src="https://user-images.githubusercontent.com/10433274/157034118-81cb305f-2913-48bf-b5cd-b16c160a63ec.png">

joebochill commented on src/blueDarkTheme.ts:
```diff
@@ -73,938 +74,992 @@ export const blueDarkTheme: ThemeOptions = {
+                        "& .MuiStepIcon-text": {
+                            fill: WhiteText,
+                            fontWeight: 600,
+                        },
```
The 'completed' styles seem to be missing for the step icon

joebochill commented on src/blueTheme.ts:
```diff
@@ -70,850 +68,938 @@ export const blueTheme: ThemeOptions = {
-            },
-            selected: {},
-            label: {},
+                selected: {},
+                label: {},
```
Can we take these out now since we don't need them?

joebochill commented on src/blueTheme.ts:
```diff
@@ -70,850 +68,938 @@ export const blueTheme: ThemeOptions = {
+                    "&:hover": {
+                        backgroundColor: Color(BLUIColors.black[500])
+                            .alpha(0.05)
+                            .string(),
+
+                    },
+                    "&.Mui-disabled": {
+                        border: `1px solid ${Color(BLUIColors.black[500])
+                            .alpha(0.12)
+                            .string()}`,
+                    },
                 },
-            },
-            groupedTextPrimary: {
-                "&:not(:last-child)$disabled": {
-                    borderColor: ThemeColors.divider,
+                contained: {
+                    "&.MuiButtonGroup-grouped": {
+                        "&:hover": {
+                            boxShadow: shadows[4],
+                        },
+                    },
```
These look like new styles that weren't there before...are they needed?

ektasawant replied:
Removed. Please review.

joebochill reviewed (commented):
I did not go through with a fine toothed comb, but the code changes look pretty good in my quick scan. As long as somebody does a side-by-side of the old with the new and things look correct, then I think we're good.

huayunh commented:
The theme transition and the input page seem to still eat up a lot of performance

# 1
Error outlined select field sometimes get the wrong color

https://user-images.githubusercontent.com/8997218/156818521-3211ea53-8e2b-40ea-ae08-e0075b675a0c.mov

# 2
Button toggle group under RTL is wrong (border on the wrong side)
![image](https://user-images.githubusercontent.com/8997218/156818731-6b1d58a2-f641-448a-8535-5c2865d92141.png)

# 3
I assume there's more to fix for the drawer rail, as the hamburger icon is misaligned
![image](https://user-images.githubusercontent.com/8997218/156818858-85167698-e2df-49f4-833d-9bd61035c9cd.png)

# 4
this is still getting the wrong color
![image](https://user-images.githubusercontent.com/8997218/156819179-ce168677-7140-4ea2-bbf0-3e78672c8e9e.png)

# 5
Chips
![image](https://user-images.githubusercontent.com/8997218/156819346-07bc0c7d-c51a-467e-8b00-9edeb0257309.png)

# 6 
I see that MUI now allows for additional component `color`s, like `color=success` etc. Can you either include those in, or maybe log an enhancement / story to include these checks for future? I logged a related design bug in #34

# 7 
This header is taking a different color
![image](https://user-images.githubusercontent.com/8997218/156819777-2cf191ec-9702-4d79-8fad-6fdb836af2a2.png)
expecting table cell / header to not have any background color

# 8 
Stepper, do not put white text on blue200. It is hard to read.
![image](https://user-images.githubusercontent.com/8997218/156821307-2ec036ce-ff09-4b88-a68d-0e51db11e15a.png)

Here's design in Figma
<img width="1026" alt="image" src="https://user-images.githubusercontent.com/8997218/156821483-0a8d9775-a866-4e9e-a092-f297b5c2cc9b.png">

# 9

I see that the surface colors are changing based on their elevation. Is this a MUI V5 thing?
![image](https://user-images.githubusercontent.com/8997218/156821708-2312fdad-cb61-410f-b827-a878e95a44f2.png)

# 10
I saw these console errors
![image](https://user-images.githubusercontent.com/8997218/156822109-1ac02984-7727-49c5-934b-26ea14cc25d6.png)

# 11
Under 2x text size, the right score card's action row starts to misalign

![image](https://user-images.githubusercontent.com/8997218/156822374-11bcc549-f206-4e96-8d60-f7b9c8616ec8.png)

ektasawant commented:
@joebochill , @huayunh will work on PR comments and update PR. Thanks for reviewing.

ektasawant commented:
@huayunh please check following:
**The theme transition and the input page seem to still eat up a lot of performance:**
It can be muiv5 issue. For reference https://github.com/mui/material-ui/discussions/29268. We can follow-up in another story.

1. **Error outlined select field sometimes get the wrong color**
Fixed
<img width="407" alt="Screenshot 2022-03-16 at 5 12 41 PM" src="https://user-images.githubusercontent.com/10433274/158582503-b1d65257-fdb3-449b-b452-2826e6dc8fb3.png">
 2 and 3 : Fixed RTL
 4:Fixed RTL
<img width="339" alt="Screenshot 2022-03-16 at 5 14 26 PM" src="https://user-images.githubusercontent.com/10433274/158582790-c923af8b-17c4-438b-bbc4-394f5c1065b8.png">
5. Fixed RTL
6. Will fix in another issue.
7.
<img width="429" alt="Screenshot 2022-03-16 at 5 16 17 PM" src="https://user-images.githubusercontent.com/10433274/158583124-1e3797cb-47ae-4416-977a-5571dc09c783.png">
8. 
<img width="829" alt="Screenshot 2022-03-16 at 5 16 57 PM" src="https://user-images.githubusercontent.com/10433274/158583215-1927ac9d-e252-43dc-8e87-b82b0de74900.png">
9. In muiv5 paper elevation having `background-image: linear-gradient(rgba(255, 255, 255, 0.07), rgba(255, 255, 255, 0.07)); `
<img width="1005" alt="Screenshot 2022-03-16 at 5 17 46 PM" src="https://user-images.githubusercontent.com/10433274/158583456-59bbae6f-7b25-4ff1-8f8a-4d02863677a6.png">
10. Fixed
11. Issue is exist in dev branch too. We can create issue for this.

huayunh commented:
I got 

> /Users/c3050144/Documents/repos/react-themes/node_modules/@mui/styled-engine/index.js
Module not found: Can't resolve '@emotion/react' in '/Users/c3050144/Documents/repos/react-themes/node_modules/@mui/styled-engine' 

When I tried running the project.

The row dividers in the table screenshot looks really dim.

Everything else looks fine from your screenshots.

ektasawant commented:
> The row dividers in the table screenshot looks really dim.

Divider color same as previous. It is looking dim because paper background.
![Uploading Screenshot 2022-03-18 at 3.33.38 PM.png…]()

For below error:
/Users/c3050144/Documents/repos/react-themes/node_modules/@mui/styled-engine/index.js
Module not found: Can't resolve '@emotion/react' in '/Users/c3050144/Documents/repos/react-themes/node_modules/@mui/styled-engine'

Did `yarn install` helped to fix this error?

huayunh commented:
The correct divider color in the dark theme should be black200 @ 36%. Is it using this color?

ektasawant commented:
> The correct divider color in the dark theme should be black200 @ 36%. Is it using this color?

Nope it was not having black200 @ 36%. Fixed it now.
<img width="1424" alt="Screenshot 2022-03-18 at 10 53 18 AM" src="https://user-images.githubusercontent.com/10433274/159045174-751d8c92-946b-458f-a457-36caf0babd56.png">

ektasawant commented:
> I got
> 
> > /Users/c3050144/Documents/repos/react-themes/node_modules/@mui/styled-engine/index.js
> > Module not found: Can't resolve '@emotion/react' in '/Users/c3050144/Documents/repos/react-themes/node_modules/@mui/styled-engine'
> 
> When I tried running the project.
> 
> The row dividers in the table screenshot looks really dim.
> 
> Everything else looks fine from your screenshots.

I tried following steps in react-themes:
1. cd demos/showcase and run `git checkout mui-v5`
2. Run `yarn start` .
3. It is working fine. Warning is known issue. https://github.com/styled-components/stylis-plugin-rtl/issues/33
<img width="924" alt="Screenshot 2022-03-21 at 8 56 45 PM" src="https://user-images.githubusercontent.com/10433274/159294300-e68af103-2d6f-4de8-ac00-781ebd4e8dc8.png">
t

huayunh commented:
Cool. The only thing I noticed was the chip border-color being off.

<img width="371" alt="image" src="https://user-images.githubusercontent.com/8997218/159358771-09916ccc-73bf-4d09-9fe5-2a4742b42f02.png">

Figma:
<img width="504" alt="image" src="https://user-images.githubusercontent.com/8997218/159358628-6f95099d-c8b1-4d54-8a57-86a820258b11.png">

huayunh commented:
> cd demos/showcase and run git checkout mui-v5

The convention across all repos has been that one should be able to run yarn start from the root, without needing to cd, checkout and yarn install. Either some commands are not right, or a yarn update submodule is needed.

huayunh commented:
#36

daileytj commented:
It looks like there is an issue with `licence-checker` a peer-dependency of `npm-license-crawler`. There's a github link showing in the package.lock that doesn't satisfy the new URL schema github is using now. Not sure what the fix is yet.

bkarambe commented on CommonOverrides.md:
```diff
@@ -11,12 +11,12 @@ If you want to override a single table:
-import { Theme } from '@material-ui/core/styles/createMuiTheme';
+import Table from "@mui/material/Table";
+import TableRow from "@mui/material/TableRow";
+import makeStyles from '@material-ui/core/styles/makeStyles';
```
This should be updated as:
import makeStyles from '@mui/styles/makeStyles';

bkarambe commented on CHANGELOG.md:
```diff
@@ -1,5 +1,7 @@
 # Changelog
-
```
Keep a blank line here for consistency.

bkarambe commented on CHANGELOG.md:
```diff
@@ -1,5 +1,7 @@
 # Changelog
-
+## v7.0.0 (Not yet published)
```
Here as well

bkarambe commented on CHANGELOG.md:
```diff
@@ -1,5 +1,7 @@
 # Changelog
-
+## v7.0.0 (Not yet published)
+### Changed
```
Same

bkarambe commented on CHANGELOG.md:
```diff
@@ -1,5 +1,7 @@
-
+## v7.0.0 (Not yet published)
+### Changed
+-   Migration from Material UI v4 to v5. [#30](https://github.com/brightlayer-ui/react-themes/issues/30)
```
Changelog text should starts with ---> Updated/Fixed/Added.
Also leave a blank line after that.

jeffvg commented:
> 

@daileytj do you have an error or something with license checker?
I did fresh clone and installed then ran generate-license with no issues.
<img width="691" alt="image" src="https://user-images.githubusercontent.com/37150565/160163700-5ba8c921-a74f-4f68-baff-44dd820371ce.png">

yarn lock has this
<img width="794" alt="image" src="https://user-images.githubusercontent.com/37150565/160163843-6369bd88-8736-410f-a93d-87ca7ce1fee6.png">

daileytj commented:
> > 
> 
> @daileytj do you have an error or something with license checker? I did fresh clone and installed then ran generate-license with no issues. <img alt="image" width="691" src="https://user-images.githubusercontent.com/37150565/160163700-5ba8c921-a74f-4f68-baff-44dd820371ce.png">
> 
> yarn lock has this <img alt="image" width="794" src="https://user-images.githubusercontent.com/37150565/160163843-6369bd88-8736-410f-a93d-87ca7ce1fee6.png">

Must have been a temporary local issue. Looks good now
